api/v1alpha1/datamigrate_types.go

```diff
@@ -0,0 +1,132 @@
+/*
+  Copyright 2023 The Fluid Authors.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// DataMigrateSpec defines the desired state of DataMigrate
+type DataMigrateSpec struct {
+	// The version information that instructs fluid to orchestrate a particular version for data migrate.
+	// +optional
+	VersionSpec `json:",inline,omitempty"`
+
+	// data to migrate source, including dataset and external storage
+	From DataToMigrate `json:"from"`
+
+	// data to migrate destination, including dataset and external storage
+	To DataToMigrate `json:"to"`
+
+	// if dataMigrate blocked dataset usage, default is false
+	// +optional
+	Block bool `json:"block,omitempty"`
+
+	// using which runtime to migrate data; if none, take dataset runtime as default
+	// +optional
+	RuntimeType string `json:"runtimeType,omitempty"`
```

Comment:
This field is only useful when `from` and `to` are both dataset for choosing runtime?

Reply:
Yes